examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,358 @@
+    # @dev contract initialization on deployment
+    # @dev will set name and symbol, interfaces, owner and URI
+    # @param name the smart contract name
+    # @param symbol the smart contract symbol
+    # @param uri the new uri for the contract
+    self.name = name
+    self.symbol = symbol
+    self.paused = False 
+    self.owner = msg.sender
+    self.uri = uri
+
+## contract status ##
+@external
+def pause():
+    # @dev Pause the contract, checks if the caller is the owner and if the contract is paused already
+    # @dev emits a pause event 
+    # @dev not part of the core spec but a common feature for NFT projects
+    assert self.owner == msg.sender, "Ownable: caller is not the owner"
+    assert not self.paused, " the contract is already paused"
+    self.paused = True
+    log Paused(msg.sender)
+
+@external
+def unpause():
+    # @dev Unpause the contract, checks if the caller is the owner and if the contract is paused already
+    # @dev emits an unpause event 
+    # @dev not part of the core spec but a common feature for NFT projects
+    assert self.owner == msg.sender, "Ownable: caller is not the owner"
+    assert self.paused, "the contract is not paused"
+    self.paused = False
+    log unPaused(msg.sender)
+
+## ownership ##
+@external
+def transferOwnership(newOwner: address):
+    # @dev Transfer the ownership. Checks for contract pause status, current owner and prevent transferring to
+    # @dev zero address
+    # @dev emits an OwnershipTransferred event with the old and new owner addresses
+    # @param newOwner The address of the new owner.
+    assert not self.paused, "The contract has been paused"
+    assert self.owner == msg.sender, "Ownable: caller is not the owner"
+    assert newOwner != self.owner, "This account already owns the contract"
+    assert newOwner != ZERO_ADDRESS, "Transfer to ZERO_ADDRESS not allowed. Use renounceOwnership() instead."
+    oldOwner: address = self.owner
+    self.owner = newOwner
+    log OwnershipTransferred(oldOwner, newOwner)
+
+@external
+def renounceOwnership():
+    # @dev Transfer the ownership to ZERO_ADDRESS, this will lock the contract
+    # @dev emits an OwnershipTransferred event with the old and new ZERO_ADDRESS owner addresses
+    assert not self.paused, "The contract has been paused"
+    assert self.owner == msg.sender, "Ownable: caller is not the owner"
+    oldOwner: address = self.owner
+    self.owner = ZERO_ADDRESS
+    log OwnershipTransferred(oldOwner, ZERO_ADDRESS)
+
+## balance ##
+@view
+@external
+def balanceOf(account: address, id: uint256) -> uint256:
+    # @dev check the balance for a specific ID and address
+    # @dev will return the balance if the id is owned by the address
+    # @dev Can also be used to check ownership of an ID
+    # @param account the address to check the balance for
+    # @param id the token ID to check the balance
+    assert account != ZERO_ADDRESS, "Please enter a valid address"
```

Comment:
The spec does not require reverting with this scenario

There is an oppurtunity to change the name of `balance` public storage variable to `balanceOf` and remove this function in favor of the public getter